lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx

```diff
@@ -0,0 +1,648 @@
+  Indices marginalSize(getDimension());
+  for (UnsignedInteger j = 0; j < getDimension(); ++ j)
+  {
+    marginalSupport[j] = marginalConditionedDistribution_.getMarginal(j).getSupport(interval.getMarginal(j));
+    marginalSize[j] = marginalSupport[j].getSize();
+  }
+  IndicesCollection tuples(Tuples(marginalSize).generate());
+  Sample support(0, getDimension());
+  for (UnsignedInteger i = 0; i < tuples.getSize(); ++ i)
+  {
+    Point x(getDimension());
+    for (UnsignedInteger j = 0; j < getDimension(); ++ j)
+      x[j] = marginalSupport[j](tuples(i, j), 0);
+    support.add(x);
+  }
+  return support;
+}
+
+/* Get one realization of the distribution */
+Point PointConditionalDistribution::getRealization() const
+{
+  if (useSimplifiedVersion_)
+    return simplifiedVersion_.getRealization();
+
+  if (distribution_.hasIndependentCopula())
+    return distribution_.getRealization().select(nonConditioningIndices_);
+  else
+    return DistributionImplementation::getRealization();
+}
+
+
+Point PointConditionalDistribution::expandPoint(const Point & point) const
+{
+  const UnsignedInteger dimension = getDimension();
+  if (point.getDimension() != dimension) throw InvalidArgumentException(HERE) << "Error: expected a point of dimension=" << dimension << ", got dimension=" << point.getDimension();
+  Point fullPoint(distribution_.getDimension());
+  for (UnsignedInteger i = 0; i < conditioningIndices_.getSize(); ++ i)
+    fullPoint[conditioningIndices_[i]] = conditioningValues_[i];
+  for (UnsignedInteger i = 0; i < dimension; ++ i)
+    fullPoint[nonConditioningIndices_[i]] = point[i];
+  return fullPoint;
+}
+
+/* Get the DDF of the distribution */
+Point PointConditionalDistribution::computeDDF(const Point & point) const
+{
+  if (useSimplifiedVersion_)
+    return simplifiedVersion_.computeDDF(point);
+  else
+    return DistributionImplementation::computeDDF(point);
+}
+
+/* Get the PDF of the distribution */
+Scalar PointConditionalDistribution::computeLogPDF(const Point & point) const
+{
+  if (useSimplifiedVersion_)
+    return simplifiedVersion_.computeLogPDF(point);
+  else
+    return distribution_.computeLogPDF(expandPoint(point)) - logNormalizationFactor_;
+}
+
+/* Get the PDF of the distribution */
+Scalar PointConditionalDistribution::computePDF(const Point & point) const
+{
+  if (useSimplifiedVersion_)
+    return simplifiedVersion_.computePDF(point);
+  else
+    return std::exp(computeLogPDF(point));
+}
+
+/* Get the CDF of the distribution */
+Scalar PointConditionalDistribution::computeCDF(const Point & point) const
+{
+  if (useSimplifiedVersion_)
+    return simplifiedVersion_.computeCDF(point);
+
+  const UnsignedInteger dimension = getDimension();
+  if (point.getDimension() != dimension)
+    throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << dimension << ", here dimension=" << point.getDimension();
+
+  Scalar cdfValue = 0.0;
+  if (distribution_.hasIndependentCopula())
```

Comment:
Same remark here: if distribution_ has an independent copula then it has a simplified version